terminal/SocketUtils.hpp

```diff
@@ -1,10 +1,23 @@
 #include "Headers.hpp"
```

Comment:
Let's rename this file to RawSocketUtils.hpp so it's more clear when to use this versus ET's sockethandler.  Also, let's create a class called RawSocketUtils and make these methods static methods of that class.